MetaMask/metamask-extension
# fix: Gas Estimation failure displays 2 duplicated Warnings

## **Description**



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23886?quickstart=1)

When there is a gas estimation failure, show only one gas warning on the confirmation screen. If the user proceeds to continue, show the gas component.

## **Related issues**

Fixes: #17866 

## **Manual testing steps**

1. Go to etherscan https://etherscan.io/address/0xe592427a0aece92de3edee1f18e0157c05861564/advanced#writeContract
2. Connect your MM acount
3. Click selfPermit
4. Fill it with values (0x, 0x for r and s)
5. Click Write
6. Should not see duplicated warnings

## **Screenshots/Recordings**



### **Before**


https://github.com/MetaMask/metamask-extension/assets/29962968/9fbee4e0-866b-4031-906e-ba2519163b3d



### **After**


https://github.com/MetaMask/metamask-extension/assets/29962968/c9e54305-e03f-42b0-923f-f8e4c641ab69



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
